### ISSUE TITLE
[request] make load_metric api intutive

### ISSUE THREAD

**Issue by stas00** [huggingface/datasets#1957]
May I suggest that `num_process` is confusing as it's singular yet expects a plural value and be deprecated in favor of `num_processes` or better to mimic the established dist environment convention for that purpose, which is `world_size`. Same for `process_id` - why reinvent the naming when we have `rank` already. That is:

```
metric = load_metric('glue', 'mrpc', world_size=world_size, rank=rank)
```

This then fits like a glove into the pytorch DDP and alike envs.

From: https://pytorch.org/docs/stable/distributed.html#torch.distributed.init_process_group

* `world_size (int, optional)` – Number of processes participating in the job. Required if store is specified.
* `rank (int, optional)` – Rank of the current process. Required if store is specified.

**Comment by mariosasko** [comment]
I agree with this proposal. IMO, `num_process` can also be misleading without reading the docs because this option may seem to leverage `multiprocessing` to compute the final result, which is not the case.

@lhoestq @albertvillanova Are you OK with breaking the API for v2.0 and renaming the params as follows:
* `num_process` -> `world_size`
* `process_id` -> `rank`

**Comment by lhoestq** [comment]
I don't think that's a good idea for 2.0, we may have a new library for metrics anyway.

Note that we will need an API that also makes sense for TF users
